redis/src/aio/connection.rs

```diff
@@ -102,7 +108,21 @@ where
 
     /// Fetches a single response from the connection.
     async fn read_response(&mut self) -> RedisResult<Value> {
-        crate::parser::parse_redis_value_async(&mut self.decoder, &mut self.con).await
+        let result = crate::parser::parse_redis_value_async(&mut self.decoder, &mut self.con).await;
+        if self.push_manager.has_sender() {
```

Comment:
this should be inverted (throughout all types, I assume) - the value is a push value, check if there's a sender. Otherwise if the user didn't set a push manager, PUSH values will be entered wrongly into the response queue.